docs/rackhd/install_os.rst

```diff
@@ -0,0 +1,116 @@
+
+Start an OS installation workflow:
+
+.. code-block:: REST
+
+    curl -X POST \
+         -H 'Content-Type: application/json' \
+         -d @params.json \
+         <server>/api/1.1/nodes/<identifier>/workflows?name=Graph.InstallCentOS
+
+
+A example of params.json with minimal parameters for installing CentOS workflow:
+
+.. literalinclude:: samples/install-os-minimal-parameters.json
+   :language: JSON
+
+A example of params.json with full set of parameters for installing CentOS workflow:
+
+.. literalinclude:: samples/install-os-maximal-parameters.json
+   :language: JSON
+
+Check the OS installation workflow is active or inactive:
+
+.. code-block:: REST
+
+    curl <server>/api/1.1/nodes/<identifier>/workflows/active
+
+
+Delete the active OS installation workflow:
+
+.. code-block:: REST
+
+    curl -X DELETE <server>/api/1.1/nodes/<identifier>/workflows/active
+
+
+OS Installation Workflow Payload
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+All parameters descriptions of OS installation workflow payload are listed below , they are fit to all supported OS installations:
+
+=================== ================ ============ ============================================
+Parameters          Type              Flags        Description
+=================== ================ ============ ============================================
+version             String           **required** The version number of target OS that needs to install. **NOTE**: For Ubuntu, the version should be 'trusty' not '14.04'.
+repo                String           **required** The OS repository address, currently only supports HTTP
+rootPassword        String           **required** The password for the OS root account
+hostname            String           **required** The hostname for target OS
+domain              String           **required** The domain for target OS
+users               Array            *optional*   If specified, this contains an array of objects, each object contains the user account information that will be created after OS installation. 0, 1, or multiple users could be specified. If *users* is omitted, null or 0 length, no user will be created.
+dnsServers          Array            *optional*   If specified, this contains an array of objects, each object contains the Domain Name Servers, multiple DNS servers could be specified, the first one will be primary, others are alternative.
+networkDevices      Array            *optional*   The static IP setting for network devices after OS installation. If it is omitted, null or 0 length, RackHD will not touch any network devices setting, so all network devices remain at the default state (usually default is DHCP).If there is multiple setting for same device, RackHD will choose the last one as the final setting, both ipv4 and ipv6 are supported here.
+rootSshKey          String           *optional*   The SSH key that will be appended into the file /root/.ssh/authorized_keys in target OS.
```

Comment:
For some OS, the `rootSshKey` may not always be stored at the `/root/.ssh/authorized_keys`. For example, it is  `/etc/ssh/keys-root/authorized_keys` for ESXi.